modules/function/src/main/java/com/opengamma/strata/function/marketdata/curve/CurveGroupMarketDataFunction.java

```diff
@@ -188,6 +203,29 @@ private CurveGroup buildGroup(
         calibratedProvider.getIndexCurves());
   }
 
+  /**
+   * Checks two values are equal and returns one of them if they are, otherwise throws an exception.
+   * <p>
+   * This is used as the merging function when collecting values into a map. If two values have the
+   * same key they must be equal. Therefore this function checks equality and returns one of the values.
+   * No actual merging is necessary as the values are the same.
+   *
+   * @param value1  a value which is being inserted into a map
+   * @param value2  a value which is being inserted into a map
+   * @return one of the values if they are equal
+   * @throws IllegalArgumentException if the values are not equal
+   */
+  private static Object checkValuesEqual(Object value1, Object value2) {
+    if (value1.equals(value2)) {
+      return value1;
+    }
+    throw new IllegalArgumentException(
+        Messages.format(
+            "Values with the same key must be equal but found unequal values: {} and {}",
+            value1,
+            value2));
```

Comment:
This error message does not contain the key, which will be painful to debug. While it is a pain, I think the calling code should be changed to not use streams in order to make the key available.